File: backend/Wms.Api/Domain/Entities/Warehouse.cs
namespace Wms.Api.Domain.Entities;

public class Warehouse
{
    public int Id { get; set; }
    public required string Code { get; set; }
    public required string Name { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public ICollection<Location> Locations { get; set; } = [];
}

File: frontend/wms-web/src/types/warehouse.ts
export type Warehouse = {
  id: number
  officeId: number
  officeName: string
  code: string
  name: string
  isActive: boolean
  createdAtUtc: string
}

export type CreateWarehouseRequest = {
  code: string
  name: string
}

export type UpdateWarehouseRequest = {
  code: string
  name: string
  isActive: boolean
}
